lib/config.ml

```diff
@@ -69,99 +69,107 @@ let of_yaml_exn str =
-     %!"
-    Fpath.pp file;
-  (match user with
-  | Some u -> Ok u
-  | None ->
+  upgrade () >>= fun () -> Ok cfg
+
+let file () = config_dir () >>| fun cfg -> Fpath.(cfg / "release.yml")
+
+let load () =
+  file () >>= fun file ->
+  OS.File.exists file >>= fun exists ->
+  if exists then OS.File.read file >>= of_yaml >>| fun x -> x else Ok empty
+
+let pretty_fields { user; remote; local; keep_v; auto_open } =
+  [
+    ("user", user);
+    ("remote", remote);
+    ("local", Stdext.Option.map ~f:Fpath.to_string local);
+    ("keep-v", Stdext.Option.map ~f:string_of_bool keep_v);
+    ("auto-open", Stdext.Option.map ~f:string_of_bool auto_open);
+  ]
+
+let save t =
+  file () >>= fun file ->
+  let fields = pretty_fields t in
+  let content =
+    let open Stdext in
+    List.filter_map fields ~f:(function
+      | _, None -> None
+      | f, Some v -> Some (Printf.sprintf "%s: %s" f v))
+  in
+  OS.File.write_lines file content
+
+let create_config ~remote_repo ~local_repo pkgs file =
+  App_log.status (fun l -> l "%a does not exist!" Fpath.pp file);
+  App_log.status (fun l ->
+      l "Please answer a few question so we can create it for you:");
+  App_log.blank_line ();
+  let guessed_user =
+    let res =
       let pkg = List.hd pkgs in
-      Pkg.infer_github_repo pkg >>= fun { owner; _ } -> Ok owner)
-  >>= fun default_user ->
-  let user = read_string default_user ~descr:"What is your GitHub ID?" in
+      Pkg.infer_github_repo pkg >>= fun { owner; _ } -> Ok owner
+    in
+    Rresult.R.to_option res
+  in
   let default_remote =
+    let open Stdext.Option.O in
     match remote_repo with
-    | Some r -> r
-    | None -> strf "git@github.com:%s/opam-repository" user
+    | Some r -> Some r
+    | None ->
+        guessed_user >|= fun user ->
+        strf "git@github.com:%s/opam-repository" user
   in
   let default_local =
     match local_repo with
-    | Some r -> Ok r
-    | None -> Ok Fpath.(v Xdg.home / "git" / "opam-repository" |> to_string)
+    | Some r -> Some r
+    | None -> Some Fpath.(v Xdg.home / "git" / "opam-repository" |> to_string)
   in
-  default_local >>= fun default_local ->
   let remote =
-    read_string default_remote
-      ~descr:
+    Prompt.user_input ?default_answer:default_remote
+      ~question:
         "What is your fork of ocaml/opam-repository? (you should have write \
          access)."
+      ()
   in
   let local =
-    read_string default_local
-      ~descr:"Where on your filesystem did you clone that repository?"
+    Prompt.user_input ?default_answer:default_local
+      ~question:"Where on your filesystem did you clone that repository?" ()
   in
   Fpath.of_string local >>= fun local ->
-  let v = strf "user: %s\nremote: %s\nlocal: %a\n" user remote Fpath.pp local in
-  OS.Dir.create Fpath.(parent file) >>= fun _ ->
-  OS.File.write file v >>= fun () ->
-  Ok
+  let config =
     {
-      user = Some user;
+      user = None;
```

Comment:
Can we go further and remove this field in this PR to avoid any mis-use?

Reply:
We probably could but I'd rather err on the safe side here, simply going through deprecation first, removing it in 2.0.